src/main/java/tools/jackson/databind/deser/jdk/CollectionDeserializer.java

```diff
@@ -217,6 +228,21 @@ public CollectionDeserializer createContextual(DeserializationContext ctxt,
         return this;
     }
 
+    protected Boolean findFormatFeatureOrClassFallback(Boolean unwrapSingle,
+        JsonFormat.Feature feature)
+    {
+        if (unwrapSingle != null) {
+            return unwrapSingle;
+        }
+        if(_classAnnotations != null){
+            JsonFormat formatAnnotation = _classAnnotations.get(JsonFormat.class);
```

Comment:
Right idea, but implementation needs bit more work: the only code that should ever access concrete annotations is `[Jackson]AnnotationIntrospector`. So we need to hold on to`AnnotatedClass`, pass that to `AnnotationIntrospector.findFormat()`.

(the reason `AnnotationIntrospector` needs to be involved is mainly for 2 things:
1. mix-in annotations
2. non-Jackson (JAXB) annotation suppor
but also gives a powerful extension point for overrides).